### PR TITLE
webapp - commit menu

### PR DESCRIPTION
**Pull request by webdevelopland**
What's new:
- new commit menu
- small design fixes of `file-changes` page

### CODE REVIEW

File: tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
+import { Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { MouseService } from '@/core/services';
+import { LoadService, StateService } from '../services';
+import { CommitInfo } from './commit-popup';
+
+// Drag element is an orange circle (or square), which can be drag and droped by user
+// to pick another commit.
+// Point is an white circle (or square), where can be dropped drag element.
```
Comment (MateMahinKhan) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts]: The comment seemed a bit funky here. I would suggest using `which can be dragged and dropped by user` and similarly in the next comment `where a dragged element (dragable?) can be dropped`

File: tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      if (this.leftDrag.isClicked) {
+        this.moveDragElement(event.pageX, this.leftDrag, this.rightDrag, -1);
+      }
+      if (this.rightDrag.isClicked) {
+        this.moveDragElement(event.pageX, this.rightDrag, this.leftDrag, 1);
+      }
+      this.moveBridge();
+    });
+
+    // When user clicks up (drops) left mouse button
```
Comment (MateMahinKhan) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts]: how about s/click up/releases?

File: tools/reviewer/webapp/src/app/dashboard/file-changes/code-changes/services/changes.service.ts [hunk 3]
```diff
@@ -125,8 +125,9 @@ export class ChangesService {
     this.addPlaceholders(textDiff.getRightChangeList(), rightBlockLines);
 
     if (leftBlockLines.length !== rightBlockLines.length) {
-      // After adding all placeholders
-      throw new Error('Blocks should have the same amount of lines');
+      throw new Error(
+        'After adding all placeholders, blocks should have the same amount of lines',
```
Comment (oferb) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/code-changes/services/changes.service.ts]: Can you add `leftBlockLines.length` and `rightBlockLines.length` to the error so we know which side has less?

File: tools/reviewer/webapp/src/app/dashboard/file-changes/code-changes/services/template.service.ts [hunk 4]
```diff
@@ -100,4 +100,16 @@ export class TemplateService {
   isEmpty(blockLine: BlockLine): boolean {
     return blockLine.threads.length === 0;
   }
+
+  isLastLine(changesLines: ChangesLine[], lineIndex: number): boolean {
```
Comment (oferb) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/code-changes/services/template.service.ts]: Can you please explain this method?
I.e why is it not simply `lineIndex === changesLines.length - 1;`

File: tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  }
+
+  moveBridge(): void {
+    this.bridge.x = this.leftDrag.x + dragWidth / 2;
+    this.bridge.width = this.rightDrag.x - this.leftDrag.x;
+  }
+
+  showPopup(commitIndex: number): void {
+    this.commitInfo.id = this.stateService.commitIdList[commitIndex];
+    // TODO: Display real time of a commit
```
Comment (oferb) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts]: Can you open an issue for this one, with a P2?

File: tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+const magnetField: number = 12;
+
+// Menu to pick commits
+// How it looks: https://i.imgur.com/Svk6ok7.png
+@Component({
+  selector: 'commit-menu',
+  templateUrl: './commit-menu.component.html',
+  styleUrls: ['./commit-menu.component.scss'],
+})
+export class CommitMenuComponent implements OnDestroy {
```
Comment (oferb) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts]: Beautiful! 👏

File: tools/reviewer/webapp/src/app/dashboard/file-changes/services/commit.service.ts [hunk 7]
```diff
@@ -48,6 +54,15 @@ export class CommitService {
 
     this.addCommitId(this.stateService.leftCommitId);
     this.addCommitId(this.stateService.rightCommitId);
+
+    // Switch left and right commits, if left commit id newer than right one
```
Comment (oferb) [inline tools/reviewer/webapp/src/app/dashboard/file-changes/services/commit.service.ts]: Can this happen through the UI, or just through the query params?

Comment (webdevelopland) [reply tools/reviewer/webapp/src/app/dashboard/file-changes/services/commit.service.ts]: Only by query params

File: tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts (hunk 6 above)
Comment (webdevelopland) [reply tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts]: :blush:

File: tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts (hunk 5 above)
Comment (webdevelopland) [reply tools/reviewer/webapp/src/app/dashboard/file-changes/commit-menu/commit-menu.component.ts]: :+1: https://github.com/google/startup-os/issues/425

File: tools/reviewer/webapp/src/app/dashboard/file-changes/code-changes/services/template.service.ts (hunk 4 above)
Comment (webdevelopland) [reply tools/reviewer/webapp/src/app/dashboard/file-changes/code-changes/services/template.service.ts]: :disappointed: round corners

**Pull request merged**